### DISCUSSION TITLE
Do webgl has it own redner loop?

### DISCUSSION THREAD

**Discussion by lv2lrn4lf** [gfxfundamentals/webgl-fundamentals#401] [General]
I was wondering if webgl has it own render loop... in the shadertoy what logic is looping the code? is it in the glsl environment to loop main function?

**Comment by greggman** [comment]
how many times main is called in a vertex shader is the `count` parameter to `gl.drawArrays(primitive, offset, count)`

how many times main is called in a fragment shader depends on what you draw. It's called once for each pixel

https://webglfundamentals.org/webgl/lessons/webgl-how-it-works.html
